### PR TITLE
run rustfmt on libcollections module

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: src/libcollections/linked_list.rs [hunk 1]
```diff
@@ -986,7 +992,7 @@ impl<A> Extend<A> for LinkedList<A> {
 }
 
 impl<I: IntoIterator> SpecExtend<I> for LinkedList<I::Item> {
-    default fn spec_extend(&mut self, iter: I) {
+    fn spec_extend(&mut self, iter: I) {
```
Comment (emilio) [inline src/libcollections/linked_list.rs]: A `default` has been stripped here.

Comment (srinivasreddy) [inline src/libcollections/linked_list.rs]: wooo. @nrc

Comment (nrc) [inline src/libcollections/linked_list.rs]: https://github.com/rust-lang-nursery/rustfmt/issues/945

File: src/libcollections/borrow.rs [hunk 2]
```diff
@@ -66,7 +66,9 @@ pub trait ToOwned {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T> ToOwned for T where T: Clone {
+impl<T> ToOwned for T
+    where T: Clone
+{
```
Comment (GuillaumeGomez) [inline src/libcollections/borrow.rs]: Why is the parenthesis going back to line? That's quite ugly. :-/

File: src/libcollections/btree/map.rs [hunk 3]
```diff
@@ -129,35 +129,38 @@ use self::Entry::*;
...
+    length: usize,
 }
 
 impl<K, V> Drop for BTreeMap<K, V> {
     #[unsafe_destructor_blind_to_params]
     fn drop(&mut self) {
         unsafe {
-            for _ in ptr::read(self).into_iter() { }
+            for _ in ptr::read(self).into_iter() {
+            }
```
Comment (GuillaumeGomez) [inline src/libcollections/btree/map.rs]: I don't find this change very useful...

File: src/libcollections/btree/map.rs [hunk 4]
```diff
@@ -244,7 +249,8 @@ impl<K, Q: ?Sized> super::Recover<Q> for BTreeMap<K, ()>
                     handle: handle,
                     length: &mut self.length,
                     _marker: PhantomData,
-                }.insert(());
+                }
+                .insert(());
```
Comment (GuillaumeGomez) [inline src/libcollections/btree/map.rs]: Not very useful once more.

File: src/libcollections/btree/map.rs [hunk 5]
```diff
@@ -311,15 +317,13 @@ pub struct RangeMut<'a, K: 'a, V: 'a> {
...
+    Vacant(#[stable(feature = "rust1", since = "1.0.0")]
+           VacantEntry<'a, K, V>),
 
     /// An occupied Entry
     #[stable(feature = "rust1", since = "1.0.0")]
-    Occupied(
-        #[stable(feature = "rust1", since = "1.0.0")] OccupiedEntry<'a, K, V>
-    ),
+    Occupied(#[stable(feature = "rust1", since = "1.0.0")]
+             OccupiedEntry<'a, K, V>),
```
Comment (GuillaumeGomez) [inline src/libcollections/btree/map.rs]: Very ugly.

File: src/libcollections/btree/map.rs [hunk 6]
```diff
@@ -962,14 +1015,15 @@ impl<K, V> IntoIterator for BTreeMap<K, V> {
...
-            length: len
+            length: len,
         }
     }
 }
 
 impl<K, V> Drop for IntoIter<K, V> {
     fn drop(&mut self) {
-        for _ in &mut *self { }
+        for _ in &mut *self {
```
Comment (GuillaumeGomez) [inline src/libcollections/btree/map.rs]: Ugly too.

**Comment by GuillaumeGomez** [comment]
As much as I like rustfmt, some cases are very ugly. :-/

File: src/libcollections/btree/map.rs (hunk 6 above)
Comment (Manishearth) [inline src/libcollections/btree/map.rs]: This should be a bug on rustfmt (I don't really find it ugly, but meh), not a comment here.

File: src/libcollections/borrow.rs (hunk 2 above)
Comment (Manishearth) [inline src/libcollections/borrow.rs]: (rustfmt bug again)

I think this is intentional though, since the where clause was indented

Comment (nrc) [inline src/libcollections/borrow.rs]: It is intentional - we do this by hand with where clauses too

**Comment by nrc** [comment]
The `default` thing has been fixed on the rustfmt side (you'll need to rebuild rustfmt from master). With that change, I think this should land.

**Comment by Manishearth** [comment]
Needs a rebase and some fixes. Closing for now, feel free to reopen if you finish these. Thanks!

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with a rebase!

**Pull request closed without merging**